src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -211,6 +211,8 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
                 itemsBuilder.Add(item);
             }
 
+            filterCache.Free();
```

Comment:
you don't free this if cancellationToken is triggered.

Reply:
fixed